src/oracle/monitor.js

```diff
@@ -0,0 +1,429 @@
+    makeRpcUtils,
+    storageHelper,
+    networkConfig
+} from '../lib/rpc.js';
+import {
+    makeFollower,
+    makeLeader,
+} from '@agoric/casting';
+import { validUrl, readJSONFile, saveJSONDataToFile } from './helper.js'
+import { getCurrent } from '../lib/wallet.js';
+import { Registry, Gauge } from 'prom-client';
+import { createServer } from 'http';
+import { parse } from 'url';
+import { iterateReverse } from '@agoric/casting';
+
+const { 
+    PORT = '3001', 
+    POLL_INTERVAL = '10', 
+    AGORIC_NET, 
+    AGORIC_RPC = "http://0.0.0.0:26657" ,
+    STATE_FILE = "data/monitoring_state.json",
+    ORACLE_FILE = "config/oracles.json",
+} = process.env;
+
+assert(!isNaN(Number(PORT)), '$PORT is required');
+assert(!isNaN(Number(POLL_INTERVAL)), '$POLL_INTERVAL is required');
+assert(validUrl(AGORIC_RPC), '$AGORIC_RPC is required');
+assert(AGORIC_NET != "" && AGORIC_NET != null, '$AGORIC_NET is required');
+
+// Create a Registry which registers the metrics
+const register = new Registry()
+
+// Add a default label which is added to all metrics
+register.setDefaultLabels({
+    app: 'agoric-cl-oracle-monitor'
+})
+
+//Create gauge for value
+const oracleSubmission = new Gauge({
+    name: 'oracle_latest_value',
+    help: 'Latest value submitted by oracle',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for timestamp
+const oracleObservation = new Gauge({
+    name: 'oracle_last_observation',
+    help: 'Last epoch in which oracle made an observation',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for last round
+const oracleLastRound = new Gauge({
+    name: 'oracle_last_round',
+    help: 'Last round in which oracle made an observation',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for price deviation
+const oracleDeviation = new Gauge({
+    name: 'oracle_price_deviation',
+    help: 'Latest price deviation by oracle',
+    labelNames: ['oracleName', 'oracle', 'feed']
+})
+
+//Create gauge for balance
+const oracleBalance = new Gauge({
+    name: 'oracle_balance',
+    help: 'Oracle balances',
+    labelNames: ['oracleName', 'oracle', 'brand']
+})
+
+//Create gauge for last price
+const actualPriceGauge = new Gauge({
+    name: 'actual_price',
+    help: 'Actual last price from feed',
+    labelNames: ['feed']
+})
+
+// Register the gaugex
+register.registerMetric(oracleSubmission)
+register.registerMetric(oracleObservation)
+register.registerMetric(oracleLastRound)
+register.registerMetric(oracleBalance)
+register.registerMetric(oracleDeviation)
+register.registerMetric(actualPriceGauge)
+
+const { agoricNames, fromBoard, vstorage } = await makeRpcUtils({ fetch });
+
+//this holds the offer ids
+var feeds = []
+//this holds the amounts in
+var amountsIn = {}
+
+/**
+ * Function to read oracles
+ * @returns oracles, their names and their addresses
+ */
+const readOracleAddresses = () => {
+    var oracles = readJSONFile(ORACLE_FILE)
+    return oracles
```

Comment:
```suggestion
    return readJSONFile(ORACLE_FILE)
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/a534c543497dd104763b1911f578cfeeb5265910